### DISCUSSION TITLE
How to deploy a monorepo to Vercel that utilizes Lerna.js scripts?

### DISCUSSION THREAD

**Discussion by sakhmedbayev** [vercel/community#2583] [Help] [answered]
I am trying to deploy a monorepo to Vercel with the following file structure:
```
packages/
 |-- db-connectors/
          |-- package.json
 |-- next-app/
          |-- package.json
package.json
```

My task is to deploy `next-app`. At the project deployment initialization, I selected `next-app` as a project root, however, I need also to run a couple of scripts in `db-connectors/package.json`. More specifically, I need to run this script `"prisma:migrate:deploy": "prisma migrate deploy"`. For that matter in the root `package.json`, I set the following script:

```
"vercel-build": "lerna run --scope=@monoreponame/db-connectors prisma:migrate:deploy && lerna run --scope=@monoreponame/next-app build --stream"
``` 

It seems that Vercel is looking for scripts only in the `next-app` directory. Is there way to run scripts that reside in the root `package.json`?

**Comment by mikestopcontinues** [comment]
How about `cd ../..` somewhere in your build package.json build script to run the other command? To clean that up, you can write a JS build script that executes multiple CLI commands in succession. Every `child_process.exec` starts from the cwd, so can write a generic script without having to worry about `cd`ing back into the specific child package.

**Reply by awitherow** [reply]
@sakhmedbayev did this end up working for you?

**Comment by amyegan** [comment, marked as answer]
Hi 👋  This discussion is a bit old, but there is some new interest in it so I want to share some info to help.

The first thing to check for Yarn and Lerna Workspaces is that the "Include source files outside of the Root Directory in the Build Step" is enabled in the Root Directory section within the project settings. It is enabled by default for all Vercel Projects created after August 27th 2020, but it is worth taking a look to make sure this is currently enabled for your Project.

I would also check that the correct Framework Preset is selected in the Build & Development Settings section of your project settings and override as needed. 

Additional resources:
- https://github.com/vercel/vercel/discussions/5431
- https://div-negi.medium.com/hosting-lerna-vercel-next-js-issues-and-headaches-with-the-process-7122e4e20202
- https://vercel.com/docs/concepts/monorepos#monorepo-faq

**Reply by grabbou** [reply]
Just wanted to say that really appreciate you point out "I would also check that the correct Framework Preset is selected in the Build & Development Settings section of your project settings and override as needed.". Such a simple yet hard to spot mistake. For some reason, my framework was "Others", and Vercel couldn't pick up the correct `dist` folder.

**Comment by justin-hackin** [comment]
My project has one web app that depends on one library within lerna. I had success with the following approach:
- leave root directory as project root
- use the following build settings:
- install: npm install && node_modules/.bin/lerna exec npm install
- build: lerna run build
- output: packages/web-app-name/dist
